### ISSUE TITLE
Multichain deployment gives EOA nonce changed unexpectedly while sending transactions

### ISSUE THREAD

**Issue by 0xTimepunk** [foundry-rs/foundry#4701]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (93016e6 2023-04-06T00:04:49.011111000Z)

### What command(s) is the bug in?

forge script

### Operating System

macOS (Apple Silicon)

### Describe the bug

When doing a multi chain local deployment, with the script pointing to localhost instances of Anvil (6 running at the same time, each forking different mainnet RPCs using quick nodes as a provider), the forge script always gives the following error:

`EOA nonce changed unexpectedly while sending transactions. Expected 3421 got 3370 from provider.), Err(
EOA nonce changed unexpectedly while sending transactions. Expected 289 got 238 from provider.)]`

The nonces with errors are always different. It seems to be the issue with two of the providers, but I'm not sure how to find which one as the logs don't say.

The script is ran with 

```
forge script script/Local.Deploy.s.sol \
    --broadcast \
    --private-key $LOCAL_PRIVATE_KEY
```

and is automatically detected as multi chain deployment.

**Comment by wrinkledeth** [comment]
I'm experiencing the same bug. Also on mac (AppleSilicon), also during multi chain local deployment.  I don't always get this error, it happens intermittently.

**Comment by ArshanKhanifar** [comment]
I'm experiencing the same bug here.

**Comment by ArshanKhanifar** [comment]
I've proposed a fix here #6271

**Comment by 0xTimepunk** [comment]
> I've proposed a fix here #6271

Lovely. Thanks for the help!

**Comment by zerosnacks** [comment]
Marking as resolved by https://github.com/foundry-rs/foundry/pull/6271
